primedev/Launcher.cmake

```diff
@@ -26,3 +26,15 @@ set_target_properties(
     NorthstarLauncher PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${NS_BINARY_DIR} LINK_FLAGS
                                                                            "/MANIFEST:NO /DEBUG /STACK:8000000"
     )
+
+if(MSVC)
+    target_compile_options(NorthstarLauncher PRIVATE /W4 /WX)
+else()
+    target_compile_options(
+        NorthstarLauncher
+        PRIVATE -Wall
+                -Wextra
+                -Wpedantic
+                -Werror
```

Comment:
Please do not make warnings fatal by default.
what is and isn't a warning can change a lot between compiler versions causing phantom errors that only appear in the CI but not locally or the other way around.

Reply:
honestly i dont think this PR is ever gonna get merged, since it should really get split up into a bunch more PRs, so I think that it's ok to have fatal warnings on this branch specifically, so that it sort of "proves" that i didnt miss any.